backend/app/api/endpoints/auth.py

```
@@ -0,0 +1,44 @@
+from core.security import ACCESS_TOKEN_EXPIRE_MINUTES, create_access_token
+from crud.user import authenticate_user, create_user, get_current_active_user
+from fastapi import APIRouter, Depends, HTTPException, status
+from fastapi.security import OAuth2PasswordRequestForm
+from schemas.user import Token, User, UserCreate
+from sqlalchemy.orm import Session
+
+router = APIRouter()
+
+
+@router.post("/register", response_model=User)
+def register_user(user: UserCreate, db: Session = Depends(get_db)):
```

Ruff (0.8.2)

Do not perform function call `Depends` in argument defaults; instead, perform the call within the function, or read the default from a module-level singleton variable

(B008)